website_sale_affiliate/models/sale_order.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    affiliate_request_id = fields.Many2one(
+        'sale.affiliate.request',
+        string='Affiliate request',
+        help='Affiliate request associated with sale order',
+    )
+
+    @api.model
+    def create(self, vals):
+        res = super(SaleOrder, self).create(vals)
+        affiliate = self.env['sale.affiliate'].find_from_session()
+        try:
+            res.affiliate_request_id = affiliate.get_request()
+        except AttributeError:
```

Comment:
> instead of one 

Whoah wait, one? In my head I see two (try, operation, except, pass) vs three (if, operation). Am I misunderstanding?

Also note that when v11 comes around, things like this will instead be replaced with:
```
with contextlib.suppress(AttributeError):
    operation()
```

Which would be the same two lines, but more Pythonic. Harder to identify what to switch during migrations if we switch to LBYL IMO